nodes/logic/evolver.py

```diff
@@ -0,0 +1,452 @@
+    fitness = 0
+    def __init__(self, genes_def, random_val=True, empty=False):
+        self.genes_def = genes_def
+        self.genes = []
+        self.fitness = 0
+        if empty:
+            return
+        if random_val:
+            for gene in genes_def:
+                agent_gene = gene.min_n + random() * gene.range
+                if gene.g_type == 'int':
+                    agent_gene = int(agent_gene)
+                self.genes.append(agent_gene)
+        else:
+            for gene in genes_def:
+                agent_gene = gene.init_val
+                self.genes.append(agent_gene)
+
+    def evaluate_fitness(self, tree, update_list, node):
+        tree.sv_process = False
+        for gen_data, agent_gene in zip(self.genes_def, self.genes):
+            if gen_data.g_type == 'float':
+                tree.nodes[gen_data.name].float_ = agent_gene
+            else:
+                tree.nodes[gen_data.name].int_ = agent_gene
+
+        tree.sv_process = True
+        do_update(update_list, tree.nodes)
+        agent_fitness = node.inputs[0].sv_get(deepcopy=False)[0]
+        if isinstance(agent_fitness, list):
+            agent_fitness = agent_fitness[0]
+        self.fitness = agent_fitness
+        # return agent_fitness
+    def cross_over(self, other_ancestor, mutation):
+        # new_agent = []
+        new_agent = DNA(self.genes_def, empty=True)
+        for g, g1, o_gene in zip(self.genes, other_ancestor.genes, self.genes_def):
+            if random() < mutation:
+                if random() < 0.5:
+                    #total gene reset
+                    new_gene = o_gene.min_n + random() * o_gene.range
+                else:
+                    #small gene mutation
+                    fp1 = random()
+                    mut = (random() - 0.5) * o_gene.range * mutation
+                    new_gene = max(min((g*fp1 + g1*(1-fp1)) + mut, o_gene.max_n), o_gene.min_n)
+
+            else:
+                fp1 = random()
+                new_gene = (g * fp1 + g1 * (1 - fp1))
+
+            if o_gene.g_type == 'int':
+                new_gene = int(new_gene)
+            new_agent.genes.append(new_gene)
+
+        return new_agent
+
+class Population:
+    population_g = []
+
+    def __init__(self, genes_def, node, tree):
+        self.node = node
+        self.node_id = node.node_id
+        self.tree = tree
+        self.iterations = node.iterations
+        self.mode = node.mode
+        self.mutation = node.mutation
+        self.fitness_booster = node.fitness_booster
+        self.time_start = time.time()
+        self.max_time = node.max_time
+        self.genes = genes_def
+        self.use_fitness_goal = node.use_fitness_goal
+        self.fitness_goal = node.fitness_goal
+        self.fitness_v = []
+        self.population_g = []
+        self.population_g.append(DNA(genes_def, random_val=False))
+        for i in range(node.population_n-1):
+            self.population_g.append(DNA(genes_def))
```

Comment:
I would not suggest to copy node attributes to the class attributes. It would be better to use this properties via a node like this: `self.node.node_id`.